### PR TITLE
Use validated call in the quantization unit test and the correct shape

### PR DESCRIPTION
**Pull request by jbobba**

### CODE REVIEW

File: test/quantize_cpu.cpp [hunk 1]
```diff
@@ -46,10 +46,10 @@ void DequantizeTest(int input, float min, float max, float expected_output)
     auto f = make_shared<Function>(r, op::ParameterVector{A});
     auto backend = runtime::Backend::create("CPU");
     // Create some tensors for input/output
-    auto a = backend->create_tensor(element::from<T>(), Shape{});
+    auto a = backend->create_tensor(element::from<T>(), Shape{1});
```
Comment (nbpatel) [inline test/quantize_cpu.cpp]: why shape{1}?

Comment (diyessi) [reply test/quantize_cpu.cpp]: `shape_a` has shape {1}. Could also make `shape_a` have shape {} and leave the others go I think.

Comment (nbpatel) [reply test/quantize_cpu.cpp]: yes, we can make shape_a have shape{} and use shape_a everywhere.  my bad that the test still has shape_a as Shape{1}

Comment (diyessi) [reply test/quantize_cpu.cpp]: There's an upcoming PR for making sure layouts assigned to a tensor agree with the shape of the tensor, and it caught this test.

Comment (jbobba) [reply test/quantize_cpu.cpp]: Shape{1} is needed to get things into MKLDNN. Currently MKLDNN doesn't like zero-dim tensors.

Comment (nbpatel) [reply test/quantize_cpu.cpp]: ah ok ya, I hit the same issue in pooling as well, they do support it in the newer release though

**Pull request merged**